### PR TITLE
Improved Atom Module to work with several links

### PR DESCRIPTION
**Pull request by eknoes**
I improved the Atom Module to work with several links, which is espacially useful for working with paged feeds like http://feeds.metaebene.me/lnp/mp3

### CODE REVIEW

File: rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java [hunk 1]
```diff
@@ -13,16 +13,19 @@
...
 
 import com.rometools.rome.feed.atom.Link;
 import com.rometools.rome.feed.module.Module;
 
+import java.util.List;
+
 public interface AtomLinkModule extends Module {
 
-    public static final String URI = "http://www.w3.org/2005/Atom";
+    String URI = "http://www.w3.org/2005/Atom";
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java]: Why did you remove `public static final`?

File: rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModuleImpl.java [hunk 2]
```diff
@@ -49,32 +52,24 @@ public String getUri() {
...
-            Link l = new Link();
-            l.setHref(link.getHref());
-            l.setType(link.getType());
-            l.setRel(link.getRel());
-            l.setHreflang(link.getHreflang());
-            l.setTitle(link.getTitle());
-            l.setLength(link.getLength());
-            m.setLink(l);
-        }
+        List<Link> result = links.subList(0, links.size());
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModuleImpl.java]: This code doesn't clone each link anymore. Not sure if it matters, but is it intentional?

File: rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java [hunk 3]
```diff
@@ -13,16 +13,19 @@
...
 
-    public Link getLink();
+    List<Link> getLinks();
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java]: This is not backward compatible, maybe keep both `get/setLinks()` and deprecated `get/setLink()`?

File: rome-modules/src/main/java/com/rometools/modules/atom/io/AtomModuleGenerator.java [hunk 4]
```diff
@@ -52,46 +54,45 @@ public final String getNamespaceUri() {
...
+                linkElement.setAttribute(hreflangAttribute);
+            }
+
+            if (link.getTitle() != null) {
+                final Attribute title = new Attribute(AtomLinkAttribute.TITLE, link.getTitle());
+                linkElement.setAttribute(title);
+            }
+
+            if (link.getLength() != 0) {
+                final Attribute lenght = new Attribute(AtomLinkAttribute.LENGTH, Long.toString(link.getLength()));
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/io/AtomModuleGenerator.java]: typo in lenght

Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/io/AtomModuleGenerator.java]: ah, it's not yours :)

File: rome-modules/src/main/java/com/rometools/modules/atom/io/AtomModuleGenerator.java [hunk 5]
```diff
@@ -52,46 +54,45 @@ public final String getNamespaceUri() {
...
-        if (link.getTitle() != null) {
-            final Attribute title = new Attribute(AtomLinkAttribute.TITLE, link.getTitle());
-            linkElement.setAttribute(title);
-        }
-
-        if (link.getLength() != 0) {
-            final Attribute lenght = new Attribute(AtomLinkAttribute.LENGTH, Long.toString(link.getLength()));
-            linkElement.setAttribute(lenght);
+    private void generateLinks(List<Link> links, Element element) {
+        for (Link link : links) {
```
Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/io/AtomModuleGenerator.java]: Maybe extract `generateLink` method? Then it will be easier to see what `generateLinks` does.

File: rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java (hunk 1 above)
Comment (PatrickGotthard) [inline rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java]: Maybe he wanted to save 3 words? 😉 Nevermind , variables in interfaces are always public static final

Comment (mishako) [inline rome-modules/src/main/java/com/rometools/modules/atom/modules/AtomLinkModule.java]: 👍

**Comment by mishako** [comment]
Hey @eknoes, any thoughts on this?

**Comment by eknoes** [comment]
Sorry, currently I don't have any time to do this. I think I will improve it in late August.

**Comment by eknoes** [comment]
Now I had the time, but because of problems with git I had to make a new PR, Sorry! See it  #320
